gnovm/pkg/gnolang/realm.go

```diff
@@ -393,14 +393,16 @@ func (rlm *Realm) FinalizeRealmTransaction(store Store) {
 func (rlm *Realm) processNewCreatedMarks(store Store, start int) int {
 	// Create new objects and their new descendants.
 	for _, oo := range rlm.newCreated[start:] {
-		if debug {
+		if debugRealm {
 			if oo.GetIsDirty() {
 				panic("new created mark cannot be dirty")
 			}
 		}
 		if oo.GetRefCount() == 0 {
-			if debug {
-				if !oo.GetIsNewDeleted() {
+			if debugRealm {
+				// The refCount for a new real object could be zero,
+				// and the object may not yet be marked as deleted.
+				if !oo.GetIsNewDeleted() && !oo.GetIsNewReal() {
```

Comment:
see zrealm_tests1.gno.
the first underlying array has refCount 0 but not yet marked new deleted yet.